view.c

```diff
@@ -144,6 +181,17 @@ view_map(struct cg_view *view, struct wlr_surface *surface)
 	}
 
 	wl_list_insert(&view->server->views, &view->link);
+
+	view->foreign_toplevel_handle = wlr_foreign_toplevel_handle_v1_create(view->server->foreign_toplevel_manager);
+	if (!view->foreign_toplevel_handle) {
+		wlr_log(WLR_ERROR, "Failed to create foreign toplevel handle");
```

Comment:
I think we can disconnect the client with `wl_resource_post_no_memory()` like is done above. This would avoid ending up with a half-constructed view.

We should make sure to `return` here to avoid accessing a NULL pointer below.